common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/DerivedImageData.scala

```diff
@@ -0,0 +1,18 @@
+package uk.ac.wellcome.models.work.internal
+
+case class DerivedImageData(
+  thumbnail: DigitalLocationDeprecated
+)
+
+object DerivedImageData {
+  def apply(image: Image[_]): DerivedImageData =
+    DerivedImageData(
+      thumbnail = image.locations
+        .find(_.locationType.id == "iiif-image")
+        .getOrElse(
+          // This should never happen
```

Comment:
If this is the case why do we account for it?

Reply:
Yeah, I can't think of a reason but if it happens then we really want to know and this custom error message is more helpful that what a plain `get` would throw. We do this elsewhere (sometimes with pattern matching) and it's been useful.